File: LeetCodes/palindrome-linked-list/1-26-2023, 10_49_36 AM/palindrome-linked-list.cpp
// Link: https://leetcode.com/problems/palindrome-linked-list

/** Submission Info:
 *
 *  AUTHOR: the_247 | Raj Patel
 *  TITLE: palindrome-linked-list
 *  RUNTIME: 346 ms
 *  MEMORY: 120.6 MB
 *  DATE: 1-26-2023, 10:49:36 AM
 *
 */

// Solution:

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        string s = "";
        while(head != NULL) {
            s += head->val;
            head = head->next;
        }
        int i = 0, j = s.length() - 1;
        while(i < j) if(s[i++] != s[j--]) return false;
        return true;
    }
};

File: LeetCodes/palindrome-linked-list/3-22-2024, 2_51_52 PM/palindrome-linked-list.java
// Link: https://leetcode.com/problems/palindrome-linked-list

/** Submission Info:
 *
 *  AUTHOR: the_247 | Raj Patel
 *  TITLE: palindrome-linked-list
 *  RUNTIME: 8 ms
 *  MEMORY: 56 MB
 *  DATE: 3-22-2024, 2:51:52 PM
 *
 */

// Solution:

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        List<Integer> list = new ArrayList<>();
        ListNode ptr = head;
        while(ptr != null) {
            list.add(ptr.val);
            ptr = ptr.next;
        }
        
        for(int i=0; i<list.size() / 2; i++) {
            if(list.get(i) != list.get(list.size() - i - 1)) {
                return false;
            }
        }
        return true;
    }
}
